gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -256,7 +258,7 @@ public static ClientContext create(StubSettings settings) throws IOException {
         .setInternalHeaders(ImmutableMap.copyOf(settings.getInternalHeaderProvider().getHeaders()))
         .setClock(clock)
         .setDefaultCallContext(defaultCallContext)
-        .setEndpoint(settings.getEndpoint())
+        .setEndpoint(endpoint)
```

Comment:
Took another look on this and I think I can revert this line change. I was concerned about the endpoint value that was passed between ClientContext <-> StubSettings and that it would lose the resolved endpoint value. The ClientContext always sets this value as the ClientSettingsEndpoint value in the EndpointContext anyways: https://github.com/googleapis/sdk-platform-java/blob/2cb0f9e542c6f074372bbaae0ab8b1f2f2598bc1/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L205.

I think if there is a need to return the resolved endpoint value from the ClientContext, I'll just create a new field called like resolvedEndpoint.